compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2043,17 +2045,39 @@ public void visit(BLangRecordLiteral recordLiteral) {
                     continue;
                 }
 
-                for (BField bField : ((BRecordType) spreadOpExpr.type).fields.values()) {
-                    if (Symbols.isOptional(bField.symbol)) {
-                        continue;
+                if (!((BRecordType) spreadOpExpr.type).sealed) {
+                    // More than one inclusive-type-descriptors with spread-fields are not allowed.
+                    if (noOfInclusiveTypes > 0) {
+                        this.dlog.error(spreadOpExpr.pos, DiagnosticCode.MORE_THAN_ONE_INCLUSIVE_TYPES,
+                                recordLiteral.expectedType.getKind().typeName());
+                    } else {
+                        noOfInclusiveTypes = noOfInclusiveTypes + 1;
+                    }
+                }
+
+                LinkedHashMap<String, BField> fieldsInRecordLiteral = ((BRecordType) spreadOpExpr.type).fields;
+                for (Object fieldName : names) {
+                    if (fieldsInRecordLiteral.containsKey(fieldName)) {
+                        if (fieldsInRecordLiteral.get(fieldName).type.tag != TypeTags.NEVER) {
+                            // An error is logged if the same field is occurred, and it is not defined as a never-type
+                            this.dlog.error(spreadOpExpr.pos, DiagnosticCode.DUPLICATE_KEY_IN_RECORD_LITERAL_SPREAD_OP,
+                                    recordLiteral.expectedType.getKind().typeName(), fieldName, spreadOpField);
+                        }
+                    } else {
+                        if (!((BRecordType) spreadOpExpr.type).sealed) {
```

Comment:
Can extract `((BRecordType) spreadOpExpr.type).sealed` out to a variable, right? Instead of accessing the `sealed` field in each iteration?

Reply:
Shall we add a comment here to say we only get here if there are other fields.

And if we don't already have one, please add a test for something like

```ballerina
public function main() {
    record {
        int x;
    } c = {x: 3, "y": 2.0};
    map<anydata> n = {...c};
}
```